typescript-generator-core/src/main/java/cz/habarta/typescript/generator/parser/Jackson2Parser.java

```diff
@@ -719,9 +719,14 @@ private DeclarationModel parseEnumOrObjectEnum(SourceType<Class<?>> sourceClass,
 
         final List<EnumMemberModel> enumMembers = new ArrayList<>();
         if (sourceClass.type.isEnum()) {
-            final Class<?> enumClass = (Class<?>) sourceClass.type;
-            final Field[] allEnumFields = enumClass.getDeclaredFields();
-            final List<Field> constants = Arrays.stream(allEnumFields).filter(Field::isEnumConstant).collect(Collectors.toList());
+            final Class<Enum> enumClass = (Class<Enum>) sourceClass.type;
+            final List<Field> constants = Arrays.stream(enumClass.getEnumConstants()).map(e -> {
```

Comment:
TBH, we don't actually need fields, the result of `enumClass.getEnumConstants()` can be used directly

Reply:
As I saw in the following code, it seems that the `Field` object is still required for the methods `Field.getAnnotation` and `public EnumMemberModel(String propertyName, String enumValue, Field originalField, List<String> comments)`. It might be quite complex to use `Enum[]` directly...

Reply:
Yes, `Field` is actually needed for getting `@Deprecated` annotation.